crates/cheatnet/src/cheatcodes/spy_events.rs

```diff
@@ -0,0 +1,69 @@
+use crate::CheatnetState;
+use cairo_felt::Felt252;
+use cairo_vm::hint_processor::hint_processor_utils::felt_to_usize;
+use starknet_api::core::ContractAddress;
+
+/// Represents emitted event. It is used in the `CheatnetState` to keep track of events
+/// emitted in the `cheatnet::src::rpc::call_contract`
+#[derive(Clone)]
+pub struct Event {
+    pub from: ContractAddress,
+    pub name: Felt252,
+    pub keys: Vec<Felt252>,
+    pub data: Vec<Felt252>,
+}
+
+/// Specifies which contract are spied on.
```

Comment:
I assume it works for events emitted by library calls? Can we include a cheatnet test for that?

Reply:
There is a problem with library calls. I noticed it yesterday and didn't have time to mention it anywhere. Basically, library calls don't return the contract address which invoked them. I wonder if it is a bug or a feature. 
However, we can deduct the address because one of the previous calls should have it. 
Will try to fix it asap